apps/openssl.c

```diff
@@ -130,23 +130,27 @@ static size_t internal_trace_cb(const char *buf, size_t cnt,
 {
     int ret = 0;
     tracedata *trace_data = vdata;
-    union {
-        CRYPTO_THREAD_ID tid;
-        unsigned long ltid;
-    } tid;
-    char buffer[256];
+    CRYPTO_THREAD_ID tid;
+    char buffer[256 + sizeof(tid)], *p;
+    size_t n = sizeof(buffer), i;
 
     switch (cmd) {
     case OSSL_TRACE_CTRL_BEGIN:
         if (!ossl_assert(!trace_data->ingroup))
             return 0;
         trace_data->ingroup = 1;
 
-        tid.ltid = 0;
-        tid.tid = CRYPTO_THREAD_get_current_id();
+        tid = CRYPTO_THREAD_get_current_id();
 
-        BIO_snprintf(buffer, sizeof(buffer), "TRACE[%lx]:%s: ", tid.ltid,
-                     OSSL_trace_get_category_name(category));
+        strncpy(buffer, "TRACE[", n);
+        p = buffer + sizeof("TRACE[") - 1;
+        n -= sizeof("TRACE[") - 1;
+        for (i = 0; i < sizeof(tid); i++) {
```

Comment:
it seems like this should be a common function:
```
CRYPTO_THREAD_to_string(CRYPTO_THREAD_ID* tid, char *buffer, size_t bufsiz)
```
Rather than three instances of the same for loop

Reply:
I considered this.  I don't want this to be an exported API, that will require ongoing support, but it is needed by apps and test.  It could be pulled into an internal header I guess but that's also a bit ugly.  I don't see a good third option :(

Reply:
I've added a commit that moves all of these conversions into an include file.